packages/react/focus-scope/src/FocusScope.tsx

```diff
@@ -103,18 +103,21 @@ const FocusScope = React.forwardRef((props, forwardedRef) => {
   React.useEffect(() => {
     if (container) {
       focusScopesStack.add(focusScope);
-      const PREVIOUSLY_FOCUSED_ELEMENT = document.activeElement as HTMLElement | null;
-
-      // we need to setup the listeners before we `dispatchEvent`
-      container.addEventListener(AUTOFOCUS_ON_MOUNT, onMountAutoFocus);
-      container.addEventListener(AUTOFOCUS_ON_UNMOUNT, onUnmountAutoFocus);
-
-      const mountEvent = new Event(AUTOFOCUS_ON_MOUNT, EVENT_OPTIONS);
-      container.dispatchEvent(mountEvent);
-      if (!mountEvent.defaultPrevented) {
-        focusFirst(getTabbableCandidates(container), { select: true });
-        if (document.activeElement === PREVIOUSLY_FOCUSED_ELEMENT) {
-          focus(container);
+      const previouslyFocusedElement = document.activeElement as HTMLElement | null;
+      const hasAutoFocusedCandidate = container.contains(previouslyFocusedElement);
+
+      if (!hasAutoFocusedCandidate) {
```

Comment:
I added a check so that if `autoFocus` has been applied to a candidate, it won't run our `onMountAutoFocus` logic. 

Without this, the container was always focused and the `autoFocus` attribute was ignored because line 118 would always be true given that the `previouslyFocusedElement` would be the `autoFocus`ed element.